src/rpc.ts

```diff
@@ -127,9 +132,12 @@ export function createRPC(
         listeners.push(() => removeEventListener(self, events.MESSAGE, handleResponse));
       }
 
-      if (guest) guest.postMessage(payload);
-      else if (isWorker() || isNodeEnv()) (self as any).postMessage(payload);
-      else (event.source || event.target).postMessage(payload, event.origin);
+      // @ts-expect-error: we know this is an array of transferables (if it exists)
+      const transferables = args[SYM_TRANSFERABLES] ?? (args.length === 1 ? args[0][SYM_TRANSFERABLES] : []);
```

Comment:
Two improvements here:

- We only have to check the `args` spread if we expect people to l `.apply` this; if that's a concern we should do it every time anyway
- We should `reduce` out the array by concat'ing from _every_ arg, vs. only looking at the first one